pysat/utils/io.py

```diff
@@ -0,0 +1,1033 @@
+                        if obj_key in loop_frame:
+                            del loop_frame[obj_key]
+
+                        # Break massive frame into bunch of smaller frames
+                        for i in np.arange(loop_lim, dtype=np.int64):
+                            loop_list.append(loop_frame.iloc[(step * i):
+                                                             (step * (i + 1)),
+                                                             :])
+                            loop_list[-1].index = new_index[(step * i):
+                                                            (step * (i + 1))]
+                            loop_list[-1].index.name = new_index_name
+                    else:
+                        loop_frame = pds.Series(loop_dict[clean_var_keys[0]],
+                                                name=obj_var_keys[0])
+
+                        # Break massive series into bunch of smaller series
+                        for i in np.arange(loop_lim, dtype=np.int64):
+                            loop_list.append(loop_frame.iloc[(step * i):
+                                                             (step * (i + 1))])
+                            loop_list[-1].index = new_index[(step * i):
+                                                            (step * (i + 1))]
+                            loop_list[-1].index.name = new_index_name
+
+                    # Add 2D object data, all based on a unique dimension within
+                    # netCDF, to loaded data dictionary
+                    loaded_vars[obj_key] = loop_list
+                    del loop_list
+
+                # Prepare dataframe index for this netcdf file
+                time_var = loaded_vars.pop(epoch_name)
+
+                # Convert from GPS seconds to seconds used in pandas (unix time,
+                # no leap)
+                # time_var = convert_gps_to_unix_seconds(time_var)
+                loaded_vars[epoch_name] = pds.to_datetime(
+                    (1.0E6 * time_var).astype(np.int64))
+                running_store.append(loaded_vars)
+                running_idx += len(loaded_vars[epoch_name])
+
+                if strict_meta:
+                    if saved_meta is None:
+                        saved_meta = meta.copy()
+                    elif (meta != saved_meta):
+                        raise ValueError(' '.join(('Metadata across filenames',
+                                                   'is not the same.')))
+
+        # Combine all of the data loaded across files together
+        out = []
+        for item in running_store:
+            out.append(pds.DataFrame.from_records(item, index=epoch_name))
+        out = pds.concat(out, axis=0)
+    else:
+        # The data is in xarray format, load differently for single or
+        # multiple files
+        if len(fnames) == 1:
+            out = xr.open_dataset(fnames[0])
+        else:
+            out = xr.open_mfdataset(fnames, combine='by_coords')
```

Comment:
```suggestion
        if len(fnames) == 1:
            out = xr.open_dataset(fnames[0], decode_timedelta=decode_timedelta)
        else:
            out = xr.open_mfdataset(fnames, combine='by_coords',
                                    decode_timedelta=decode_timedelta)
```

Reply:
The bugfix here got dropped in the copy, but the tests to check this did get merged in, hence the errors for `test_utils.py::TestLoadNetCDF4XArray::test_read_netcdf4_with_time_meta_labels`

Reply:
The kwarg `decode_time_delta` needs to be added to the functions and documentation.  I think this branch may predate #886.  See changes there for example

Reply:
Fixed!